dan_layer/consensus/src/hotstuff/on_propose.rs

```diff
@@ -235,6 +242,21 @@ where TConsensusSpec: ConsensusSpec
             commands.iter().map(|c| c.to_string()).collect::<Vec<_>>().join(",")
         );
 
+        let non_local_buckets = proposer::get_non_local_buckets_from_commands(
+            tx,
+            &commands,
+            local_committee_shard.num_committees(),
+            local_committee_shard.bucket(),
+        )?;
+
+        let mut foreign_indexes = HashMap::new();
+
+        for non_local_bucket in non_local_buckets {
+            foreign_indexes
+                .entry(non_local_bucket)
+                .or_insert(foreign_counters.increment_counter(non_local_bucket));
```

Comment:
This will run eagerly (which I assume is unintentional).

```suggestion
                .or_insert_with(|| foreign_counters.increment_counter(non_local_bucket));
```